test/unit/synopsys-action/diagnostics.test.ts

```diff
@@ -0,0 +1,38 @@
+import * as configVariables from '@actions/artifact/lib/internal/config-variables'
+import {tmpdir} from 'os'
+import {uploadDiagnostics} from '../../../src/synopsys-action/diagnostics'
+
+const fs = require('fs')
+import * as artifact from '@actions/artifact'
+
+import * as inputs from '../../../src/synopsys-action/inputs'
+import {UploadResponse} from '@actions/artifact'
+
+let tempPath = '/temp'
+
+beforeEach(() => {
+  tempPath = tmpdir()
+  Object.defineProperty(process, 'platform', {
+    value: 'linux'
+  })
+})
+
+test('Test uploadDiagnostics expect API error', () => {
```

Comment:
Shall we include positive test cases as well?

Reply:
done